models/stg_klaviyo__event.sql

```diff
@@ -35,7 +35,13 @@ rename as (
         cast(person_id as {{ dbt.type_string() }} ) as person_id,
         type,
         uuid,
-        property_value as numeric_value,
+        {% if target.type == 'bigquery' %}
+            cast(regexp_replace(cast(property_value as {{ dbt.type_string() }}), r'[^0-9.]*', '') as {{ dbt.type_numeric() }}) as numeric_value,
+        {% elif target.type == 'postgres' %}
+            cast(regexp_replace(cast(property_value as {{ dbt.type_string() }}), '[^0-9.]*', '', 'g') as {{ dbt.type_numeric() }}) as numeric_value,
+        {% else %}
+            cast(regexp_replace(cast(property_value as {{ dbt.type_string() }}), '[^0-9.]*', '') as {{ dbt.type_numeric() }}) as numeric_value,
+        {% endif %}
```

Comment:
Can consider making this a macro in the package (to then possibly move to fivetran_utils) instead of leveraging conditionals?

Reply:
Macro created, let me know if it all looks good!